### PR TITLE
V ghajam/rest samples refactor

### PR DESCRIPTION
**Pull request by GhiathAjam**
~# Will be merged into 'main' \*\*after\*\* #14~
~### Diffing now to latest #14~
#14 was merged and now diffing into `main`

#### PR Classification
This pull request **refactors** REST code samples code by updating comments and introducing minor changes

#### PR Summary
- Updating old/inaccurate comments.
- Updating `.gitignore` to ignore `python` temp cache and virtual environment files.
- Pretty printing `response.headers`.
- Using `print(json.dumps(response.json(), indent=4))` instead of the buggy `pprint(response.json())`.
- Removing the need for `urllib.parse`.
- Adding timeouts to `request.(get|post)`.

### CODE REVIEW

File: samples/rest/BingVisualSearchV7.py [hunk 1]
```diff
@@ -1,46 +1,49 @@
-# Download and install Python at https://www.python.org/ 
+# Download and install Python at https://www.python.org/
 # Run the following in a command console window: pip3 install requests
```
Comment (kerolloz) [inline samples/rest/BingVisualSearchV7.py:2]: the comment in line 2 is a bit outdated, it should rely on the requirements.txt file

File: samples/rest/BingSpellCheckV7.py [hunk 2]
```diff
@@ -1,53 +1,51 @@
-# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (kerolloz) [inline samples/rest/BingSpellCheckV7.py:1]: Why are we removing the copyright comment

**Review by kerolloz — COMMENTED** [review]
I would recommend specifying, maybe in the README file, what linter we're using, if any, and also it would be great to add a GitHub action to enforce subsequent PRs to follow the same style. If you want to go the extra mile, you can add a .vscode/ directory with 2 files:
- `extensions.json` it should include the recommended extensions to use for development, such as Pylint, or flake8. 
- `settings.json` it should include things like autoFormatOnSave using the extensions to ensure consistent code style for future contributors. 

you could take a look on https://github.com/kerolloz/kounter/tree/master/.vscode as an example.

File: samples/rest/BingSpellCheckV7.py (hunk 2 above)
Comment (arwong) [reply samples/rest/BingSpellCheckV7.py:1]: Copyright headers are an org requirement.

File: samples/rest/BingVisualSearchV7.py [hunk 3]
```diff
@@ -1,46 +1,49 @@
...
 
 # Add your Bing Visual Search V7 subscription key to your environment variables / .env file
 subscription_key = os.environ.get(SUBSCRIPTION_KEY_ENV_VAR_NAME)
 if subscription_key is None:
-    raise(RuntimeError(f'Please define the {SUBSCRIPTION_KEY_ENV_VAR_NAME} environment variable'))
+    raise (RuntimeError(
+        f'Please define the {SUBSCRIPTION_KEY_ENV_VAR_NAME} environment variable'))
 
 image_path = './my_image.jpg'
+image_path = 'C:\\Users\\v-ghajam\\Pictures\\black.jpg'
```
Comment (arwong) [inline samples/rest/BingVisualSearchV7.py:28]: Remove test code

File: samples/rest/BingVisualSearchV7.py (hunk 1 above)
Comment (GhiathAjam) [reply samples/rest/BingVisualSearchV7.py]: Done e9ef3aa

File: samples/rest/BingVisualSearchV7.py (hunk 3 above)
Comment (GhiathAjam) [reply samples/rest/BingVisualSearchV7.py]: Done e9ef3aa

**Review by Mohammed-Deifallah — APPROVED** [review]

File: samples/rest/BingSpellCheckV7.py (hunk 2 above)
Comment (GhiathAjam) [reply samples/rest/BingSpellCheckV7.py:1]: Done 673fabd

**Comment by GhiathAjam** [comment]
@kerolloz  Thanks for the wonderful suggestions. I'll add a Linter action and a Tester action on a later PR.

File: samples/rest/BingSpellCheckV7.py [hunk 4]
```diff
@@ -1,53 +1,45 @@
...
-# Licensed under the MIT License.
-
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.import json
```
Comment (arwong) [inline samples/rest/BingSpellCheckV7.py:2]: >import json [](http://example.com/codeflow?start=33&length=11)

Seems like you got an extra "import json" in all of these

**Review by arwong — APPROVED** [review]
:shipit:

File: samples/rest/BingSpellCheckV7.py (hunk 4 above)
Comment (GhiathAjam) [reply samples/rest/BingSpellCheckV7.py]: Done c1e833d
`BingImageSearchV7.py` had an extra `import os`

File: samples/rest/BingCustomSearchV7.py [hunk 5]
```diff
@@ -1,38 +1,39 @@
...
-# <request>
-r = requests.get(url, headers={AUTH_HEADER_NAME: subscription_key})
-pprint(json.loads(r.text))
-# </request>
+params = {'q': searchTerm, 'customconfig': customConfigId}
+
+
+url = 'https://api.bing.microsoft.com/v7.0/custom/search'
+
+response = requests.get(
```
[the remark this answers is not in the document]
Comment (GhiathAjam) [reply samples/rest/BingCustomSearchV7.py]: Done f779faf

**Review by kerolloz — APPROVED** [review]
Great work ✨

**Pull request merged**